### PR TITLE
Add legacy application token section

### PR DESCRIPTION
**Pull request by mpmadhavig**
### Purpose
This PR will add a Legacy Application token sention to maintain the below configs.
1. useClientIdAsSubClaimForAppTokens
2. omitUsernameInIntrospectionRespForAppTokens

The initial UI design is as follows.
Screenshot 2024-08-16 at 08 34 38

Please feel free to make suggestions for the design.

### Related Issues
- https://github.com/wso2/product-is/issues/20917

### Checklist

- [ ] e2e cypress tests locally verified. (for internal contributers)
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Relevant backend changes deployed and verified
- [ ] [Unit tests](/docs/testing/UNIT_TESTING.md) provided. (Add links if there are any)
- [ ] [Integration tests](https://github.com/wso2/product-is/tree/master/modules/integration) provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines
- [ ] Ran FindSecurityBugs plugin and verified report
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets

### CODE REVIEW

File: features/admin.applications.v1/components/forms/inbound-oidc-form.tsx [hunk 1]
```diff
@@ -2566,6 +2576,130 @@ export const InboundOIDCForm: FunctionComponent<InboundOIDCFormPropsInterface> =
...
+                                        </AlertTitle>
+                                        <Trans
+                                            i18nKey={ t("actions:fields.authentication.info.message") }
+                                        >
+                                            You currently using an outdated behavior for application tokens.
+                                            Please follow the below guideline before migrating to the new behavior.
+                                            <ol>
+                                                <li>
+                                                    <strong>Client Application Changes:</strong>
+                                                    <p>Update your client application to no longer use the
```
Comment (Achintha444) [inline features/admin.applications.v1/components/forms/inbound-oidc-form.tsx:2610]: `use the` needs space after.

**Comment by Achintha444** [comment]
**Suggestion for the design:**

- reduce the gap between the title and the warning.
- increase the gap between the checkbox and the warning.
- review the content with the docs team.

File: features/admin.applications.v1/components/forms/inbound-oidc-form.tsx [hunk 2]
```diff
@@ -290,6 +295,8 @@ export const InboundOIDCForm: FunctionComponent<InboundOIDCFormPropsInterface> =
     const requestObjectEncryptionMethod: MutableRefObject<HTMLElement> = useRef<HTMLElement>();
     const subjectToken: MutableRefObject<HTMLElement> = useRef<HTMLElement>();
     const applicationSubjectTokenExpiryInSeconds: MutableRefObject<HTMLElement> = useRef<HTMLElement>();
+    const useClientIdAsSubClaimForAppTokensEle: MutableRefObject<HTMLElement> = useRef<HTMLElement>();
+    const omitUsernameInIntrospectionRespForAppTokensEle: MutableRefObject<HTMLElement> = useRef<HTMLElement>();
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/components/forms/inbound-oidc-form.tsx:299]:
```suggestion
    const useClientIdAsSubClaimForAppTokensElement: MutableRefObject<HTMLElement> = useRef<HTMLElement>();
    const omitUsernameInIntrospectionRespForAppTokensElement: MutableRefObject<HTMLElement> = useRef<HTMLElement>();
```

File: features/admin.applications.v1/components/forms/inbound-oidc-form.tsx [hunk 3]
```diff
@@ -2566,6 +2576,130 @@ export const InboundOIDCForm: FunctionComponent<InboundOIDCFormPropsInterface> =
...
+                                (
+                                    <Alert severity="warning">
+                                        <AlertTitle
+                                            className="alert-title">
+                                            <Trans components={ { strong: <strong/> } } >Note:</Trans>
+                                        </AlertTitle>
+                                        <Trans
+                                            i18nKey={ t("actions:fields.authentication.info.message") }
+                                        >
+                                            You currently using an outdated behavior for application tokens.
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/components/forms/inbound-oidc-form.tsx:2605]:
```suggestion
                                            You are currently using an outdated behavior for application tokens.
```

Comment (pavinduLakshan) [inline features/admin.applications.v1/components/forms/inbound-oidc-form.tsx:2605]: Using `you` in UI instructions is generally not recommended. Hope we can get the text reviewed with the doc team.

cc: @himeshsiriwardana

File: features/admin.applications.v1/components/forms/inbound-oidc-form.tsx [hunk 4]
```diff
@@ -2566,6 +2576,130 @@ export const InboundOIDCForm: FunctionComponent<InboundOIDCFormPropsInterface> =
...
+                                            Please follow the below guideline before migrating to the new behavior.
+                                            <ol>
+                                                <li>
+                                                    <strong>Client Application Changes:</strong>
+                                                    <p>Update your client application to no longer use the 
+                                                        <code>sub</code> claim to refer to the application
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/components/forms/inbound-oidc-form.tsx:2611]: `claim` is too technical for an average user. we can use `attribute` instead.

File: features/admin.applications.v1/components/forms/inbound-oidc-form.tsx [hunk 5]
```diff
@@ -2566,6 +2576,130 @@ export const InboundOIDCForm: FunctionComponent<InboundOIDCFormPropsInterface> =
...
+                                (
+                                    <Alert severity="warning">
+                                        <AlertTitle
+                                            className="alert-title">
+                                            <Trans components={ { strong: <strong/> } } >Note:</Trans>
+                                        </AlertTitle>
+                                        <Trans
+                                            i18nKey={ t("actions:fields.authentication.info.message") }
+                                        >
+                                            You are currently using an outdated behavior for application tokens.
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/components/forms/inbound-oidc-form.tsx:2605]: Better if we can get the text reviewed by the doc team?

**Comment by mpmadhavig** [comment]
This will track from https://github.com/wso2/identity-apps/pull/6813

**Pull request closed without merging**